ext/pdo/pdo_stmt.c

```diff
@@ -725,23 +749,54 @@ static bool do_fetch(pdo_stmt_t *stmt, zval *return_value, enum pdo_fetch_type h
 	how = how & ~PDO_FETCH_FLAGS;
 
 	if (!do_fetch_common(stmt, ori, offset)) {
-		return 0;
+		return false;
 	}
 
 	if (how == PDO_FETCH_BOUND) {
 		RETVAL_TRUE;
-		return 1;
-	}
-
-	if ((flags & PDO_FETCH_GROUP) && stmt->fetch.column == -1) {
-		colno = 1;
-	} else {
-		colno = stmt->fetch.column;
+		return true;
 	}
 
 	if (how == PDO_FETCH_LAZY) {
 		get_lazy_object(stmt, return_value);
-		return 1;
+		return true;
+	}
+
+	/* When fetching a column we only do one value fetch, so handle it separately */
```

Comment:
This could still be in the `switch`, no?

Reply:
Because of the weird special case with `PDO_FETCH_GROUP` (which probably should also take into account FETCH_UNIQUE now that I think about it) I'd rather leave this outside the switch statement.

A follow-up I think to do is instead of the code setting `stmt->fetch.column == -1` prior to calling `do_fetch()` it should set `stmt->fetch.column == 0` and let the new grouping code handle it. Probably requires having a "common" `colno` variable however.